fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryStreamingAggregationPlan.java

```diff
@@ -346,14 +346,34 @@ public static RecordQueryStreamingAggregationPlan of(@Nonnull final Quantifier.P
         final var groupingKeyAlias = CorrelationIdentifier.uniqueID();
         final var aggregateAlias = CorrelationIdentifier.uniqueID();
 
-        return new RecordQueryStreamingAggregationPlan(inner,
+        return RecordQueryStreamingAggregationPlan.of(inner,
                 groupingKeyValue,
                 aggregateValue,
                 groupingKeyAlias,
                 aggregateAlias,
                 completeResultValueSupplier.supply(groupingKeyValue, aggregateValue, groupingKeyAlias, aggregateAlias));
     }
 
+    @Nonnull
+    public static RecordQueryStreamingAggregationPlan of(@Nonnull final Quantifier.Physical inner,
```

Comment:
I think this is the best way of doing this. Do get into the hair of flattened result vs nested result.